hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -183,6 +183,12 @@ func (r *NodePoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.HostedCluster, nodePool *hyperv1.NodePool) (ctrl.Result, error) {
 	log := ctrl.LoggerFrom(ctx)
 
+	// The nodeCount field got renamed, copy the old field for compatibility purposes if the
+	// new one is unset.
+	if nodePool.Spec.Replicas == nil {
```

Comment:
what it nodeCount is changed afterwards? should we enforce unset it? should we log it'd be ignored?

Reply:
We can not always copy, that would make it impossible to update `Replicas`